module-network.tf

```diff
@@ -98,8 +100,7 @@ module "network" {
   create_cluster               = var.create_cluster
   create_bastion               = var.create_bastion
   create_fss                   = var.create_fss
-  create_nsgs                  = var.create_nsgs
-  create_nsgs_always           = var.create_nsgs_always
+  nsgs                         = var.nsgs
```

Comment:
I probably asked this before but I'll ask again just to clarify:

- `never` and `always` are self-explanatory
- what does auto do?

Reply:
>Defaults to automatic creation when ... resources are expected to utilize it.
I added notes in the variable descriptions for these to help explain - `auto` will use logic to determine if creation is deemed necessary, e.g. if operator creation is disabled, `auto` would disable the associated subnet/NSG/etc. when used. This was the standard behavior previously, and `never`/`always` provide additional control when creating only network resources/using existing ones.

Regarding the `nsgs` var change: this better aligns with the structure used for `subnets`, with more granular control over each element.